core/dbt/graph/queue.py

```diff
@@ -37,7 +37,8 @@ def __init__(
         # this lock controls most things
         self.lock = threading.Lock()
         # store the 'score' of each node as a number. Lower is higher priority.
-        self._scores = self._calculate_scores()
+        # TODO: incorporate _include_in_cost (or remove dead code, still needed?)
+        self._scores = {y: x for x, y in enumerate(nx.topological_sort(self.graph))}
```

Comment:
This is code we haven't touched for some time, perhaps since https://github.com/fishtown-analytics/dbt/pull/1157.

https://blog.getdbt.com/how-we-made-dbt-runs-30--faster/

> This is a key insight, and it’s the most meaningful performance optimization in dbt today. Unfortunately, the logic required to intelligently parallelize the execution of the DAG is complicated, and we can’t just throw topological sort at this problem. To understand why, consider the output of the topological sort algorithm. Suppose we had two threads ready to start executing models. The first thread can definitely get started on model A, but what should the second thread do? It can’t process model K until model A has completed, but that information isn’t present in the topological ordering of the DAG. Topological sort just isn’t applicable in a multi-threaded use case.

Does this still hold true?

Reply:
Updated!